### ISSUE TITLE
RedisNodeNotFoundException: Node: /xxx.xxx.xxx.xxx:xxx for slot: 15242 hasn't been discovered yet

### ISSUE THREAD

**Issue by hjhhjhhjh** [redisson/redisson#944]
## This is log print.......

```
2017-06-29 10:46:27.948 WARN [redisson-netty-1-11][io.netty.util.internal.logging.Slf4JLogger:151] - An exception was thrown by org.redisson.command.CommandAsyncService$9.operationComplete()
org.redisson.client.RedisNodeNotFoundException: Node: /10.1.21.4:7002 for slot: 15242 hasn't been discovered yet
        at org.redisson.connection.MasterSlaveConnectionManager.getEntry(MasterSlaveConnectionManager.java:715) ~[redisson-3.4.3.jar!/:?]
        at org.redisson.connection.MasterSlaveConnectionManager.connectionWriteOp(MasterSlaveConnectionManager.java:705) ~[redisson-3.4.3.jar!/:?]
        at org.redisson.command.CommandAsyncService.async(CommandAsyncService.java:502) ~[redisson-3.4.3.jar!/:?]
        at org.redisson.command.CommandAsyncService.checkAttemptFuture(CommandAsyncService.java:754) ~[redisson-3.4.3.jar!/:?]
        at org.redisson.command.CommandAsyncService.access$300(CommandAsyncService.java:79) ~[redisson-3.4.3.jar!/:?]
        at org.redisson.command.CommandAsyncService$9.operationComplete(CommandAsyncService.java:609) ~[redisson-3.4.3.jar!/:?]
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:500) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:479) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:420) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:122) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at org.redisson.misc.RedissonPromise.tryFailure(RedissonPromise.java:98) [redisson-3.4.3.jar!/:?]
        at org.redisson.client.protocol.CommandData.tryFailure(CommandData.java:78) [redisson-3.4.3.jar!/:?]
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:247) [redisson-3.4.3.jar!/:?]
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:127) [redisson-3.4.3.jar!/:?]
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489) [netty-codec-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:367) [netty-codec-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265) [netty-codec-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:134) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) [netty-transport-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138) [netty-common-4.1.11.Final.jar!/:4.1.11.Final]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
```
## This is jstack print......
```
"http-nio-8080-exec-25" #2698 daemon prio=5 os_prio=0 tid=0x00007f6d30009000 nid=0x5790 waiting on condition [0x00007f6d3ede3000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x000000076ecb1968> (a java.util.concurrent.CountDownLatch$Sync)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
        at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
        at org.redisson.command.CommandAsyncService.get(CommandAsyncService.java:147)
        at org.redisson.RedissonObject.get(RedissonObject.java:70)
        at org.redisson.RedissonBucket.set(RedissonBucket.java:131)
```

###### Whenever this problem occurs, restarting application will resolve this error

i had talked to Nikita Koksharov 
and i think the problem is that 
when redis cluster update metadata, redisson can't be noticed right away,then it read the old metadata,but why the thread be stopped?
please help

**Comment by jackygurui** [comment]
The reason thread stops is because you are calling the sync method. As you can clearly see from the stack trace, under the hood, it parks the thread by using a count down latch and wait for the command to return, because you have a problem with running that command, in this situation, the thread will be parked until the timeout happens.

**Comment by hjhhjhhjh** [comment]
then you mean the error occurs by redisson itself? it's not caused by our operation?

**Comment by jackygurui** [comment]
What I mean is this is how synced methods work, that thread been parked not a problem or error, it's an expected behaviour.

**Comment by hjhhjhhjh** [comment]
I understand.
Now the situation is never come out.
I would like to know is my configuration problem or redisson internal problems

**Comment by jackygurui** [comment]
So can you explain how the problem had happened?

**Comment by mrniko** [comment]
@hjhhjhhjh 
> but why the thread be stopped?

Fixed in https://github.com/redisson/redisson/issues/1048
